### ISSUE TITLE
Koalas option fails resulting in Exception

### ISSUE THREAD

**Issue by gsheni** [alteryx/woodwork#541]
- An EvalML user reported this [error](https://github.com/alteryx/evalml/issues/1780), which is caused by Woodwork.
- We should look into making it so that if the `ks.set_option` cannot be safely called, `woodwork` import does not fail.
  - OR We should also look into if `set_option` can only be called when using a Koalas DataFrame with Woodwork (instead of being located at the top of a file and outside a function).

```python
~\Anaconda3\lib\site-packages\evalml\demos\breast_cancer.py in
1 import pandas as pd
----> 2 import woodwork as ww
3 from sklearn.datasets import load_breast_cancer as load_breast_cancer_sk
4
5

~\Anaconda3\lib\site-packages\woodwork_init_.py in
1 # flake8: noqa
2 from .config import config
----> 3 from .datatable import DataColumn, DataTable
4 from .type_sys import type_system
5 from .type_sys.utils import list_logical_types, list_semantic_tags

~\Anaconda3\lib\site-packages\woodwork\datatable.py in
27 ks = import_or_none('databricks.koalas')
28 if ks:
---> 29 ks.set_option('compute.ops_on_diff_frames', True)
```

**Comment by dsherry** [comment]
@gsheni @thehomebrewnerd wooo, you guys are fast!

Between when I responded to https://github.com/alteryx/evalml/issues/1780 and now I worked up a draft PR which I believe will fix this problem. I'll get that ready for review. If you agree, let's merge it. If not, I will gladly cede the issue back to you @thehomebrewnerd  :)

**Comment by dsherry** [comment]
@gsheni @thehomebrewnerd 

My PR #542 will prevent the error reported in https://github.com/alteryx/evalml/issues/1780 from causing the woodwork import to crash. I think its a helpful change because it'll a) prevent this issue from crashing import in woodwork and dependent projects (evalml) and b) give us more info if users encounter similar issues in the future. However, given the stack trace from https://github.com/alteryx/evalml/issues/1780, #542 is probably not enough to close this issue.

My suggestion: merge #542 and get that into evalml, keep this issue open and try to reproduce the error with `koalas.set_option` from https://github.com/alteryx/evalml/issues/1780

**Comment by thehomebrewnerd** [comment]
@dsherry I agree that we should at least spend a little more time trying to reproduce the error. However, my gut feel is that this is a result of an issue with the user's koalas setup, or perhaps a koalas bug as the only thing that happened to generate the original error was importing koalas and setting the koalas option. 

In any case I will investigate further and see if I can reproduce, and I agree that in the meantime we can implement a fix to prevent the error for users that are not using Koalas.

**Comment by thehomebrewnerd** [comment]
Reopening this for now, pending additional investigation of the original root cause.

**Comment by dsherry** [comment]
@thehomebrewnerd @gsheni I don't think #543 is enough to either a) avoid the user-reported error in https://github.com/alteryx/evalml/issues/1780 or b) fix the underlying error (which I think we agree is probably dependent on the user's env and we may not be able to do much about).

#543 simply moved the code in question from import-time to `DataTable` init-time. If users encounter the error from https://github.com/alteryx/evalml/issues/1780, they will now be unable to initialize a `DataTable`, instead of being unable to import woodwork entirely. While this is an improvement, I don't think its enough. This is why I added a `try`/`except` in #542 

Given that we have a user reporting a stack trace coming up through the call to `koalas.set_option`, I think we should write our code to expect that stack trace could occur, and be ready to make woodwork still function even if that error is thrown. Otherwise, anyone with this particular issue will be blocked from using woodwork and dependent libraries (evalml, featuretools).

Do you agree?

**Comment by thehomebrewnerd** [comment]
@dsherry I agree that moving the code will not avoid the error. However, as I mentioned elsewhere, this setting this config is **not optional** for users to use Koalas dataframes with Woodwork.

If we use a try/except block, the option will not get set if it fails, and Woodwork will subsequently fail in another spot downstream, which will further obscure the fact the the failure was caused by this option not getting set.

Bottom line, we can't make this `set_option` call optional if a user is working with Koalas. What we can do, and what #543 does, is prevent the error from causing an import failure if a user has Koalas installed in their environment, but is not working with Koalas dataframes.

**Comment by dsherry** [comment]
@thehomebrewnerd ah, understood, I didn't realize that config setting is a requirement for us! Thanks for clarifying.

**Comment by thehomebrewnerd** [comment]
@dsherry @gsheni I have tried unsuccessfully to reproduce the original error from [EvalML](https://github.com/alteryx/evalml/issues/1780) in Python 3.6, 3.7 and 3.8. I have always been able to import evalml and woodwork, both with and without koalas and pyspark installed. This was tested using Woodwork 0.0.7 which is currently pinned by evalml.

Not sure what else to do at this point without additional information from the original user. I am still suspecting this has something to do with the Spark environment setup, and not a bug in Woodwork, but I also don't have anything to know that for sure.

Any other ideas?

**Comment by gsheni** [comment]
I am closing this issue and until we get more [information](https://github.com/alteryx/evalml/issues/1780#issuecomment-773557075) from the end-user.
